apm-agent-plugins/apm-javalin-plugin/src/main/java/co/elastic/apm/agent/javalin/JavalinInstrumentation.java

```diff
@@ -0,0 +1,163 @@
+            final boolean isLambdaHandler = handlerClassName.equals("co.elastic.apm.agent.javalin.JavalinHandlerLambdaInstrumentation$WrappingHandler");
+
+            // transaction name gets only set if we are dealing with a HTTP method processing, not before/after handlers
+            if (ctx.handlerType().isHttpMethod()) {
+                final StringBuilder name = transaction.getAndOverrideName(PRIO_HIGH_LEVEL_FRAMEWORK);
+                if (name != null) {
+                    transaction.setFrameworkName(FRAMEWORK_NAME);
+                    transaction.setFrameworkVersion(VersionUtils.getVersion(Handler.class, "io.javalin", "javalin"));
+                    transaction.withType("request");
+
+                    name.append(ctx.handlerType().name()).append(" ").append(ctx.endpointHandlerPath());
+
+                    // no need for anonymous handler class names in the transaction
+                    if (!isLambdaHandler) {
+                        name.append(" ").append(handlerClassName);
+                    }
+                }
+            }
+
+            // do not create an own span for JavalinServlet.addHandler, as this not added by the users code and leads to confusion
+            if (handlerClassName.startsWith("io.javalin.http.JavalinServlet")) {
+                return null;
+            }
+
+            // create own span for all handlers including after/before
+            return startSpan(handlerClassName, ctx.handlerType(), isLambdaHandler, ctx.matchedPath(), ctx.method(), ctx.url());
+        }
+
+        @Advice.OnMethodExit(suppress = Throwable.class, onThrowable = Throwable.class, inline = false)
+        public static void onAfterExecute(@Advice.Enter @Nullable Object spanObj,
+                                          @Advice.Argument(0) Object ctxObject,
+                                          @Advice.Thrown @Nullable Throwable t) {
+            if (spanObj != null) {
+                Context ctx = ((Context) ctxObject);
+                final Span span = (Span) spanObj;
+                span.deactivate();
+
+                final CompletableFuture<?> responseFuture = ctx.resultFuture();
+                if (responseFuture == null) {
+                    // sync request
+                    span.captureException(t).end();
+                } else {
+                    // future was set in the handler, so we need to end the span only on future completion
+                    responseFuture.whenComplete((o, futureThrowable) -> span.captureException(futureThrowable).end());
+                }
+            }
+        }
+
+        private static Span startSpan(String handlerClassName, HandlerType handlerType, boolean isLambdaHandler, String matchedPath, String method, String url) {
+            final AbstractSpan<?> parent = tracer.getActive();
+            if (parent == null) {
+                return null;
+            }
+
+            Span span = parent.createSpan().activate();
```

Comment:
[minor] I prefer to see the `activate` in the `onMethodEnter` advice, it's easier to follow the state with regard to deactivation in `onMethodExit`.

Reply:
changed this method to only set the span name so all the code remains in the advice